flycheck.el

```diff
@@ -2159,10 +2162,43 @@ into the verification results."
         (princ (make-string (- message-column (current-column)) ?\ ))
         (let ((message (flycheck-verification-result-message result))
               (face (flycheck-verification-result-face result)))
-          (insert (propertize message 'face face)))
+          ;; If face is nil, using propertize erases the face already contained
+          ;; by the message.  We don't want that, since this would remove the
+          ;; button face from the checker chain result.
+          (if face
+              (insert (propertize message 'face face))
+            (insert message)))
         (princ "\n"))))
   (princ "\n"))
 
+(defun flycheck-checker-chain (checker)
+  "Return the chain of checkers following CHECKER.
+
+The return value is a list of checkers, not including CHECKER."
+  (let ((chain))
+    (while (setq checker (flycheck-get-next-checker-for-buffer checker))
+      (push checker chain))
+    (nreverse chain)))
```

Comment:
Do we only want the usable ones here? Wouldn't it better instead to include also the ones that may not be used for this buffer? (`-get-next-checker-for-buffer` calls `flycheck-may-use-next-checker`, which checks the current max error level, etc)

Reply:
Hmm the problem with the static view of available next checkers is that it is a tree, not a list.

We could display the immediate next checkers of the current one; that is easy and not ambiguous.

Reply:
Indeed :thinking:
I wonder if the simplest wouldn't be to just show the list of next checkers for each checker.